### PR TITLE
[Enhancement] Input Validation for generate obo token endpoint

### PR DESCRIPTION
**Pull request by RyanL1997**
### Description
Add Input Validation for generate obo token endpoint
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
Enhancement

### Issues Resolved
* Resolve https://github.com/opensearch-project/security/issues/3558
* Relate https://github.com/opensearch-project/security/issues/2573

### Testing
Integration Test

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/integrationTest/java/org/opensearch/security/http/OnBehalfOfJwtAuthenticationTest.java [hunk 1]
```diff
@@ -68,9 +69,16 @@ public class OnBehalfOfJwtAuthenticationTest {
     public static final String OBO_USER_NAME_NO_PERM = "obo_user_no_perm";
     public static final String DEFAULT_PASSWORD = "secret";
     public static final String NEW_PASSWORD = "testPassword123!!";
-    public static final String OBO_TOKEN_REASON = "{\"reason\":\"Test generation\"}";
+    public static final String OBO_TOKEN_REASON = "{\"description\":\"Test generation\"}";
     public static final String OBO_ENDPOINT_PREFIX = "_plugins/_security/api/generateonbehalfoftoken";
     public static final String OBO_DESCRIPTION = "{\"description\":\"Testing\", \"service\":\"self-issued\"}";
+
+    public static final String OBO_DESCRIPTION_WITH_INVALID_DURIATIONSECONDS =
```
Comment (DarshitChanpura) [inline src/integrationTest/java/org/opensearch/security/http/OnBehalfOfJwtAuthenticationTest.java:76]: nit: `OBO_DESCRIPTION_WITH_INVALID_DURATIONSECONDS`

File: src/main/java/org/opensearch/security/action/onbehalf/CreateOnBehalfOfTokenAction.java [hunk 2]
```diff
@@ -176,4 +191,29 @@ private Set<String> mapRoles(final User user) {
...
+            return (Integer) durationObj;
+        } else if (durationObj instanceof String) {
+            try {
+                return Integer.parseInt((String) durationObj);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("durationSeconds must be an integer.");
+            }
+        } else {
+            throw new IllegalArgumentException("durationSeconds must be an integer.");
+        }
```
Comment (DarshitChanpura) [inline src/main/java/org/opensearch/security/action/onbehalf/CreateOnBehalfOfTokenAction.java:217]:
```suggestion
if (durationObj instanceof Integer) {
    return (Integer) durationObj;
} else if (durationObj instanceof String) {
    try {
        return Integer.parseInt((String) durationObj);
    } catch (NumberFormatException ignored) {
    }
}
throw new IllegalArgumentException("durationSeconds must be an integer.");
```

**Review by DarshitChanpura — COMMENTED** [review]
Thank you @RyanL1997 for this improvement. Left some minor suggestions.

File: src/integrationTest/java/org/opensearch/security/http/OnBehalfOfJwtAuthenticationTest.java (hunk 1 above)
Comment (RyanL1997) [reply src/integrationTest/java/org/opensearch/security/http/OnBehalfOfJwtAuthenticationTest.java]: good catch! fixed

File: src/main/java/org/opensearch/security/action/onbehalf/CreateOnBehalfOfTokenAction.java (hunk 2 above)
Comment (RyanL1997) [reply src/main/java/org/opensearch/security/action/onbehalf/CreateOnBehalfOfTokenAction.java]: applied

**Review by DarshitChanpura — APPROVED** [review]

**Review by cwperks — APPROVED** [review]

**Pull request merged**